tests/e2e/metamask.spec.ts

```diff
@@ -117,6 +118,44 @@ test('deploy a token', async ({ page }) => {
   await expect(page.locator('#tokenAddress')).toContainText(/0x.+/)
 })
 
+const getTransactionCount = async (page: Page, account: string): Promise<number> => {
+  const res = await page.evaluate(addr =>
+    window.ethereum.request({
+      method: 'eth_getTransactionCount',
+      params: [addr, 'latest']
+    }),
+    account
+  )
+  return Number(res)
+}
+
+test('send legacy transaction', async ({ page, accounts }) => {
+  await page.locator('text=Connect').click()
+  await wallet.authorize(Web3RequestKind.RequestAccounts)
+
+  const nonceBefore = await getTransactionCount(page, accounts[0])
+
+  await page.locator('text=Send Legacy Transaction').click()
+  await wallet.authorize(Web3RequestKind.SendTransaction)
```

Comment:
Let's get `nonceBefore` right before authorizing the request, so we will be sure that transaction is not sent before it.